lib/tty/prompt/answers_collector.rb

```diff
@@ -44,11 +48,23 @@ def create_collector
 
       # @api public
       def add_answer(answer)
-        @answers[@name] = answer
+        current = @answers[@name]
+
+        if current.nil? || !@multiple
```

Comment:
I think this conditional is a sign that the dry has been taken a step too far. 

 I would probably revert back and tackle `key` and `values` separately without the `@multiple` key. The `values` call should always ensure that we're dealing with array for a given `name`:
```
@answers[@name] = [@answer[@name]] unless @answer[@name].is_a?(Array)
```
Having done that, the `add_answer` needs to only figure out if it needs to append or override:
```
if @answers[@name].is_a?(Array)
  @answers[@name] = answer
else
  @answers[@name] << answer
end
```